File: FLUTTER/igest_app/lib/models/principal/sede.dart
import 'package:google_maps_flutter/google_maps_flutter.dart';

class Sede {
  int idSede;
  int idPadre;
  int idEmpresa;
  int nivel;
  int idTipoSede;
  String nomSede;
  String dirSede;
  String idDistrito;
  bool activo;
  bool sunatOficial;
  String sunatCod;
  String facEleRuta;
  String facEleToken;
  String latitud;
  String longitud;
  int orden;
  String nomImagen;
  LatLng coords;

  Sede({required this.idSede, required this.idPadre, required this.idEmpresa, required this.nivel, required this.idTipoSede, required this.nomSede, required this.dirSede, required this.idDistrito, required this.activo, required this.sunatOficial, required this.sunatCod, required this.facEleRuta, required this.facEleToken, required this.latitud, required this.longitud, required this.orden, required this.nomImagen, required this.coords});
  // Constructor reducido
  Sede.mini({required this.idSede, required this.idEmpresa, required this.nomSede, required this.dirSede, required this.nomImagen, required this.coords}) : idPadre = -1, nivel = -1, idTipoSede = -1, idDistrito = '', activo = false, sunatOficial = true, sunatCod = '', facEleRuta = '', facEleToken = '', latitud = '', longitud = '', orden = -1;

  // Sede.fromJson(Map<String, dynamic> json)
  //   : idSede = json['idSede'] ?? -1,
  //     idPadre = json['idPadre'] ?? -1,
  //     idEmpresa = json['idEmpresa'] ?? -1,
  //     nivel = json['nivel'] ?? 0,
  //     idTipoSede = json['idTipoSede'] ?? -1,
  //     nomSede = json['nomSede'] ?? '',
  //     dirSede = json['dirSede'] ?? '',
  //     idDistrito = json['idDistrito'] ?? '',
  //     activo = json['activo'] ?? false,
  //     sunatOficial = json['sunatOficial'] ?? false,
  //     sunatCod = json['sunatCod'] ?? '',
  //     facEleRuta = json['facEleRuta'] ?? '',
  //     facEleToken = json['facEleToken'] ?? '',
  //     latitud = json['latitud'] ?? '',
  //     longitud = json['longitud'] ?? '',
  //     orden = json['orden'] ?? 0,
  //     nomImagen = json['nomImagen'] ?? '',
  //     coords = LatLng(double.tryParse(json['latitud'] ?? '0') ?? 0.0, double.tryParse(json['longitud'] ?? '0') ?? 0.0);

  // Map<String, dynamic> toJson() => {'idSede': idSede, 'idPadre': idPadre, 'idEmpresa': idEmpresa, 'nivel': nivel, 'idTipoSede': idTipoSede, 'nomSede': nomSede, 'dirSede': dirSede, 'idDistrito': idDistrito, 'activo': activo, 'sunatOficial': sunatOficial, 'sunatCod': sunatCod, 'facEleRuta': facEleRuta, 'facEleToken': facEleToken, 'latitud': latitud, 'longitud': longitud, 'orden': orden, 'nomImagen': nomImagen, 'coords': coords.toJson()};
}

File: SPRING/igest.server/src/main/java/com/yupanastudio/igest/server/principal/model/Sede.java
/*
 * Document   : Sede
 * Author     : Franklin Barzola Olivares <frankoantonio@gmail.com>
 */
package com.yupanastudio.igest.server.principal.model;

import org.springframework.data.annotation.Id;

public class Sede {

    @Id
    private Integer idSede;
    private int idPadre;
    private int idEmpresa;
    private int nivel;
    private int idTipoSede;
    private String nomSede;
    private String dirSede;
    private String idDistrito;
    private Boolean activo;
    private Boolean sunatOficial;
    private String sunatCod;
    private String facEleRuta;
    private String facEleToken;
    private String latitud;
    private String longitud;
    private int orden;

    public Integer getIdSede() {
        return idSede;
    }

    public void setIdSede(Integer idSede) {
        this.idSede = idSede;
    }

    public int getIdPadre() {
        return idPadre;
    }

    public void setIdPadre(int idPadre) {
        this.idPadre = idPadre;
    }

    public int getIdEmpresa() {
        return idEmpresa;
    }

    public void setIdEmpresa(int idEmpresa) {
        this.idEmpresa = idEmpresa;
    }

    public int getNivel() {
        return nivel;
    }

    public void setNivel(int nivel) {
        this.nivel = nivel;
    }

    public int getIdTipoSede() {
        return idTipoSede;
    }

    public void setIdTipoSede(int idTipoSede) {
        this.idTipoSede = idTipoSede;
    }

    public String getNomSede() {
        return nomSede;
    }

    public void setNomSede(String nomSede) {
        this.nomSede = nomSede;
    }

    public String getDirSede() {
        return dirSede;
    }

    public void setDirSede(String dirSede) {
        this.dirSede = dirSede;
    }

    public String getIdDistrito() {
        return idDistrito;
    }

    public void setIdDistrito(String idDistrito) {
        this.idDistrito = idDistrito;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    public Boolean getSunatOficial() {
        return sunatOficial;
    }

    public void setSunatOficial(Boolean sunatOficial) {
        this.sunatOficial = sunatOficial;
    }

    public String getSunatCod() {
        return sunatCod;
    }

    public void setSunatCod(String sunatCod) {
        this.sunatCod = sunatCod;
    }

    public String getFacEleRuta() {
        return facEleRuta;
    }

    public void setFacEleRuta(String facEleRuta) {
        this.facEleRuta = facEleRuta;
    }

    public String getFacEleToken() {
        return facEleToken;
    }

    public void setFacEleToken(String facEleToken) {
        this.facEleToken = facEleToken;
    }

    public String getLatitud() {
        return latitud;
    }

    public void setLatitud(String latitud) {
        this.latitud = latitud;
    }

    public String getLongitud() {
        return longitud;
    }

    public void setLongitud(String longitud) {
        this.longitud = longitud;
    }

    public int getOrden() {
        return orden;
    }

    public void setOrden(int orden) {
        this.orden = orden;
    }

}
